File: app/src/main/java/com/yuk/miuiHomeR/ui/WidgetActivity.java
package com.yuk.miuiHomeR.ui;

import androidx.fragment.app.Fragment;

import com.yuk.miuiHomeR.R;
import com.yuk.miuiHomeR.ui.base.BaseAppCompatActivity;
import com.yuk.miuiHomeR.ui.base.SubFragment;

public class WidgetActivity extends BaseAppCompatActivity {

    @Override
    public Fragment initFragment() {
        return new WidgetFragment();
    }


    public static class WidgetFragment extends SubFragment {

        @Override
        public int getContentResId() {
            return R.xml.prefs_widget;
        }

        @Override
        public void initPrefs() {

        }
    }
}

File: app/src/main/java/com/yuk/miuiHomeR/ui/WidgetActivity.kt
package com.yuk.miuiHomeR.ui

import androidx.fragment.app.Fragment
import com.yuk.miuiHomeR.R
import com.yuk.miuiHomeR.ui.base.BaseAppCompatActivity
import com.yuk.miuiHomeR.ui.base.SubFragment

class WidgetActivity : BaseAppCompatActivity() {

    override fun initFragment(): Fragment {
        setTitle(R.string.widget)
        return WidgetFragment()
    }

    class WidgetFragment : SubFragment() {
        override fun getContentResId(): Int {
            return R.xml.prefs_widget
        }

        override fun initPrefs() {}
    }
}